lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -90,15 +183,57 @@ module.exports.prototype = {
     },
 
     check: function(file, errors) {
-        file.iterateNodesByType('BlockStatement', function(node) {
-            var openingBrace = file.getFirstNodeToken(node);
-            var prevToken = file.getPrevToken(openingBrace);
+        var setting = this._setting;
+        if (setting === true) {
+            file.iterateNodesByType('BlockStatement', function(node) {
+                var openingBrace = file.getFirstNodeToken(node);
+                var prevToken = file.getPrevToken(openingBrace);
 
-            errors.assert.sameLine({
-                token: prevToken,
-                nextToken: openingBrace,
-                message: 'Newline before curly brace for block statement is disallowed'
+                errors.assert.sameLine({
+                    token: prevToken,
+                    nextToken: openingBrace,
+                    message: 'Newline before curly brace for block statement is disallowed'
+                });
             });
-        });
+        } else {
+            file.iterateNodesByType(['BlockStatement'], function(node) {
+                var type = getBlockType(node);
+                if (setting.indexOf(type) !== -1) {
```

Comment:
`if (setting === true || setting.indexOf(type) !== -1) {` and then the outer if/else goes away.